### DISCUSSION TITLE
text-base not being overwritten by screen size properties

### DISCUSSION THREAD

**Discussion by UndercoverNL** [tailwindlabs/tailwindcss#12580] [Help] [answered]
Hi there, today I ran in to a issue on my React Tailwind project, where I was trying to increase a font size on larger screen sizes, in specifically I tried to increase a `text-base` to a `sm:text-2x1`, which didn't seem to work in any case. Then I tried it with a `text-sm` it worked normally, which makes me think this is a bug within tailwind not overwritten the `text-base` class correctly.

It happens for all screen sizes, so `text-base` to `lg:text-3x1` also had this issue.

![image](https://github.com/tailwindlabs/tailwindcss/assets/42718013/a9d33ee8-0626-4797-abe0-3734554f97cc)

The exact testing code I used:
```
<h2 className="text-green text-base sm:text-2xl font-semibold">Title</h2>
```

If anyone knows a solution for this, I'd appreciate it.

**Comment by wongjn** [comment]
You seem to have two distinct CSS files, `output.css` and `App.css`. To ensure the Tailwind classes work as expected, consider only having one CSS file that compiles Tailwind classes. Alternatively, if one is from a library and another is from your project, ensure your project CSS file is linked in the HTML *after* the library CSS file.

**Reply by UndercoverNL** [reply]
The App.css only contains

```
@tailwind base;
@tailwind components;
@tailwind utilities;
```

And the output file is the generated tailwind file, which doesn't have any manual changes of course

**Reply by wongjn** [reply]
It seems like you may have some misconfiguration somewhere, consider providing a remotely-hosted git repository that reproduces the unexpected behavior.

**Reply by UndercoverNL** [reply]
https://github.com/Arteeks-Web-Solutions/permacultuurleertuinen

**Reply by wongjn** [reply, marked as answer]
As mentioned in my original post, you seem to have two distinct CSS files, `output.css` and `App.css` `import`ed in `src/App.js`:

```js
import './App.css';
import './dist/output.css';

// …
```

It seems like `src/dist/output.css` is some precompiled Tailwind CSS which is thus conflicting with your own Tailwind compilation. Consider removing the `import` (and perhaps file):

```diff
 import './App.css';
-import './dist/output.css';
 import { Route, BrowserRouter, Routes } from 'react-router-dom';
 import Home from "./home/Home.js";
```

**Reply by UndercoverNL** [reply]
the output.css file is the tailwind generated file which is noted in the installation documentation: https://tailwindcss.com/docs/installation

If I remove that import tailwind will be removed and all tailwind classes will not work anymore.

**Reply by wongjn** [reply]
All of that is incorrect.

https://tailwindcss.com/docs/installation is when using the Tailwind CLI. You are using `react-scripts` via Create React App, of which the relevant instructions are at https://tailwindcss.com/docs/guides/create-react-app.

Tailwind would not be removed and all Tailwind classes will work *even better* than they do currently (as long as your `content` file globs are correct). This can be evidenced by the fact that we can see that class name rules are already being compiled from `App.css` as per your initial screenshot.

**Reply by UndercoverNL** [reply]
aa that is my bad I thought the CLI would need to be used for a react app too, anyway, if I remove the import / file a lot of my css gets messed up, looks like color specific classes like `bg` do not work anymore, that is what made me think that the classes where removed.

**Reply by wongjn** [reply]
`bg` would not be a valid class with your Tailwind configuration.

When you remove the import / file and your CSS gets "messed up" is probably because of misconfiguration. I can see only one class related to color, `text-green` in `src/App.js` which should work. Colors from Tailwind's default color palette will not work since you configured your custom colors in `theme.colors` as per [the documentation](https://tailwindcss.com/docs/customizing-colors#using-custom-colors). If you wanted to keep most of Tailwind's default color palette , consider [modifying your custom colors declarations to be at `theme.extend.colors` instead](https://tailwindcss.com/docs/customizing-colors#adding-additional-colors):

```diff
 theme: {
-  colors: {
-    'green': '#38902B',
-  },
-  extend: {},
+  extend: {
+    colors: {
+      'green': '#38902B',
+    },
+  },
 },
```

**Reply by UndercoverNL** [reply]
got it, thanks for your quick responses, appreciate the help.
